proposed/ObjectToObjects.md

```diff
@@ -0,0 +1,56 @@
+Object To Objects
+=================
+
+Summary
+-------
+
+Objects are always passed by reference, not by value. This means you always have shared state when passing objects around.
+Shared state is one of the main culprits for unnecessary or incidental complexity. Complexity is bad.
+
+re-But...tals
+-------------
+
+- "But I use value objects!" Good for you, that is not actually OOP though.
+
+- "But all my objects are immutable." Yup, same as value objects. See above.
+
+- "OOP is the future!". Carnegie Melloin disagrees, they don't actually offer OOP as a course anymore. OO Design does 
+  have merit. But you should probably check out this Functional Programming thing.
+
+In the wild
+-----------
+
+- DateTime, part of PHP. There is now also an ImmutableDateTime. Guess why.
+
+- PSR-7: URL class ->withMethod('OPTIONS') - this is good btw.
+
+Elaboration
+-----------
+
+Complexity is the main problem with software development. It is always better to keep things simple. Simple is not easy.
+It may be more difficult to understand or longer to type. But a simple solution means less ties, less connections to
+other parts of a program. We already know this. That is why global variables are considered bad, any part of your program
+can change one and a totally unrelated part ( or so you thought ) will suddenly behave differently. This is also why we
+use dependency injection ( note: not service locators, there is a difference ).
+
+By using basic types PHP gives you a great boon, called 'pass by value'. So if I call a function with a variable and the 
```

Comment:
"_By using_ basic types PHP you _get_ a great boon" or "_Using_ basic types PHP _gives_ you a great boon".